lib/flappy_search_bar.dart

```diff
@@ -340,83 +396,117 @@ class _SearchBarState<T> extends State<SearchBar<T>>
   }
 
   Widget _buildContent(BuildContext context) {
+    Widget content;
+
     if (_error != null) {
-      return _error;
+      content = _error;
     } else if (_loading) {
-      return widget.loader;
+      content = widget.loader;
     } else if (_searchQueryController.text.length < widget.minimumChars) {
       if (widget.placeHolder != null) return widget.placeHolder;
-      return _buildListView(
+      content = _buildListView(
           widget.suggestions, widget.buildSuggestion ?? widget.onItemFound);
     } else if (_list.isNotEmpty) {
-      return _buildListView(_list, widget.onItemFound);
+      content = _buildListView(_list, widget.onItemFound);
     } else {
-      return widget.emptyWidget;
+      content = widget.centerEmptyWidget
+          ? Center(child: widget.emptyWidget)
+          : widget.emptyWidget;
     }
+
+    return Padding(
+      padding: widget.contentPadding,
+      child: content,
+    );
   }
 
   @override
   Widget build(BuildContext context) {
+    if (_firstLaunch && widget.autoFocus) {
+      /// Gives search bar focus on first build.
+      _firstLaunch = false;
+      FocusScope.of(context).requestFocus(_searchQueryFocusNode);
+    }
+
     final widthMax = MediaQuery.of(context).size.width;
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: <Widget>[
-        Padding(
-          padding: widget.searchBarPadding,
-          child: Container(
-            height: 80,
-            child: Row(
-              mainAxisAlignment: MainAxisAlignment.start,
-              children: <Widget>[
-                Flexible(
-                  child: AnimatedContainer(
-                    duration: Duration(milliseconds: 200),
-                    width: _animate ? widthMax * .8 : widthMax,
-                    decoration: BoxDecoration(
-                      borderRadius: widget.searchBarStyle.borderRadius,
-                      color: widget.searchBarStyle.backgroundColor,
-                    ),
-                    child: Padding(
-                      padding: widget.searchBarStyle.padding,
-                      child: Theme(
-                        child: TextField(
-                          controller: _searchQueryController,
-                          onChanged: _onTextChanged,
-                          style: widget.textStyle,
-                          decoration: InputDecoration(
-                            icon: widget.icon,
-                            border: InputBorder.none,
-                            hintText: widget.hintText,
-                            hintStyle: widget.hintStyle,
+        Container(
+          decoration: BoxDecoration(
+            color: widget.searchBarStyle.surroundingColor,
+          ),
+          child: Padding(
+            padding: widget.searchBarPadding,
+            child: Container(
+              height: 80,
```

Comment:
Could you remove this hard coded value too please? If the user want to expand the Container for some reason, he could use `widget.searchBarPadding`
Is there any reason why this value has been hard-coded?

Reply:
Hi, which hard-coded value are you referring to? If you are referring to `height: 80`, that was previously there. What I have done in this part was to only wrap the entire widget with a `Container`, allowing users to add a background color to the area surrounding the search bar.

Reply:
Yes I'm talking about the `height: 80`. I know it was already there, but since you are editing this part I was wondering if you could remove it.
This fixed value lets the Container that wrap the SearchBar having a fixed height and is not nice in some layout situations

Reply:
The surroundingArea that will be filled with `surroundingColor` could be rendered giving a `widget.searchBarPadding`. This way if no Padding value is provided the real Bar element (the Row) will take the whole Container and the total height will be `widget.searchBarStyle.searchBarHeight`.

Right now even with zero Padding and even if I set a `widget.searchBarStyle.searchBarHeight` of 40 for example, I'll have the Row taking 80.0 on height

Reply:
Will spend some time to do this in a few days time. Will appreciate any help from anyone to edit in the mean time!

Reply:
@dariof28 changes are done. Do you have write access in this repo?

Reply:
Unfortunately no :pensive:

Reply:
That 80 height is actually a problem for me right now! Thanks for your solution! Lets hope changes will be merged soon...